openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraEchoMethodContract.scala

```diff
@@ -0,0 +1,180 @@
+package com.linagora.openpaas.james.common
+
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.core.ResponseObject.SESSION_STATE
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, BOB, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder}
+import org.apache.james.utils.DataProbeImpl
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+trait LinagoraEchoMethodContract {
+
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString())
+      .addUser(BOB.asString(), BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .build
+  }
+
+  @Test
+  def echoMethodShouldRespondOKWithRFC8621VersionAndSupportedMethod(): Unit = {
+    val request =
+      """{
+        |  "using": [
+        |    "urn:ietf:params:jmap:core"
+        |  ],
+        |  "methodCalls": [
+        |    [
+        |      "Linagora/echo",
```

Comment:
shouldn't we add a capability to allow us to accept this new method? I thought this was discussed in an other PR but I don't see it fixed

Reply:
Should we though if it's meant to be temporary?

Reply:
we need to make changes to james-project code for it to work i think... So in a different PR may be?

Reply:
You should not add custom linagora capability into the apache james code I believe

Reply:
But I'm pretty sure with the latest refactorings on jmap methods it should be generic enough to allow us to do it from openpaas-james. Let me check...

Reply:
Then we need to extends the JmapApiRoutes with our own class and override the JMAP module in all 3 servers. All that work deserves another PR?

Reply:
We should have a pluggable mechanism to inject capabilities into JMAP sessions <3

I'm ok to have it as follow up PRs.

Reply:
Ok... @LanKhuat is right, we do define the supported capabilities hardcoded in apache james, making it impossible to add new custom ones from outside of it... That would require a smart code change in the James base indeed

Reply:
See https://github.com/linagora/openpaas-james/issues/28

https://issues.apache.org/jira/browse/JAMES-3490